ssl/s3_clnt.c

```diff
@@ -684,7 +684,8 @@ int ssl3_client_hello(SSL *s)
     buf = (unsigned char *)s->init_buf->data;
     if (s->state == SSL3_ST_CW_CLNT_HELLO_A) {
         SSL_SESSION *sess = s->session;
-        if ((sess == NULL) || (sess->ssl_version != s->version) ||
+        if ((sess == NULL) ||
+            !ssl23_version_supported(s, sess->ssl_version) ||
```

Comment:
This doesn't seem to make sense.  I think only SSLv3 is supported in this file and so you'll just end up checking just the same.